CDCgov/prime-reportstream
# Engagement/angela/8087/load option for multiple settings

This PR adds additional options to the `./prime multiple-settings get` cli command. Specifically `-l, --load-to-local` to automatically load the settings with transport modified to be SFTP to the local database, and `-a, --append-to-orgs` to append the settings to the organizations.yml file. They can be used independently of each other. When used together, the `-a, --append-to-orgs` option will append the settings with the modified transport to the organizations.yml file.

### Test steps
1. Navigate to https://github.com/CDCgov/prime-reportstream in a browser
2. Click the Code button
3. Copy the location
4. In a terminal navigate to the location you want the repository in
5. Run the command `git clone past_stuff_from_step_3_here`
6. Run the command `git checkout engagement/angela/8087/load_option_for_multiple-settings`
7. Follow instructions here for setup if necessary https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/getting-started/getting-started.md
8. In a terminal navigate to the prime-router directory
9. Run the command `./prime login --env staging`
10. Enter your Okta preview (ReportStream staging environment) username and password in the window that pops up
11. Confirm with your MFA
12. Close the window
13. Return to the terminal
14. Navigate to the prime-router directory if necessary
15. Run the command `./prime multiple-settings get --filter wa-phd --env staging`
16. Save the output for reference later
17. Run the command `./prime multiple-settings get --filter wa-phd`
18. Confirm that this returns a different result than the staging environment
19. Save the output for reference later
20. Run the command ` ./prime multiple-settings get --output ./junk/waphd.yml --filter wa-phd -a --env staging`
21. Confirm that the same settings for wa-phd in the staging environment are output to ./junk/waphd.yml
22. Confirm that settings for wa-phd are appended to the organizations.yml file with the transport modified to 

> transport
    type: SFTP
    host: sftp
    port: 22
    filePath: ./upload
    credentialName: DEFAULT-SFTP

23. Run the command `./prime multiple-settings get --filter wa-phd`
24. Confirm that this output has not changed
25. Save the changes to organizations.yml elsewhere to be referenced later
26. Roll back changes to organizations.yml
27. Run the command `./prime multiple-settings get --filter wa-phd --env staging -l`
28. Confirm that organizations.yml has not changed
29. Run the command `./prime multiple-settings get --filter wa-phd`
30. Confirm that the output is the same as the output you saved in step 25 and not the same as the output from step 19
31. Run the command `./gradlew resetDB`
32. Run the command `./gradlew reloadTables`
33. Run the command `./gradlew reloadSettings`
34. Run the command `./prime multiple-settings get --filter wa-phd --env staging -l -a`
35. Confirm that organizations.yml has been updated in the same manner as before
36. Run the command `./prime multiple-settings get --filter wa-phd`
37. Confirm that the output matches the update to organizations.yml
38. Run the command `./gradlew resetDB`
39. Run the command `./gradlew reloadTables`
40. Run the command `./gradlew reloadSettings`
41. Run the command ` ./prime multiple-settings get --output ./junk/waphd.yml --filter wa-phd -l -a --env staging`
42. Confirm that the same settings for wa-phd in the staging environment are output to ./junk/waphd.yml
43. Confirm that organizations.yml has been updated in the same manner as before
44. Run the command `./prime multiple-settings get --filter wa-phd`
45. Confirm that the output matches the update to organizations.yml
46. Run the command `./prime multiple-settings get --help`
47. Verify that there are entries for `-l, --load-to-local` and `-a, --append-to-orgs` and that they contain useful information

### Testing

## Linked Issues
- Fixes #8087
